examples/equilibria/fem_fixed_boundary/fem_magnetostatic_2D_single_coil.py

```diff
@@ -0,0 +1,269 @@
+
+# create the coil (rectangular cross section) and set the mesh options
+# Note: just to simulate a "real" application, enclosure is generated into the xz
+# plane, but then its placement is changed in order to be in the xy plane (due to a
+# limitation in the importing of the mesh with msh2xdmf for which only the first 2
+# column are considered when importing a 2D mesh)
+
+# %%
+
+poly_coil = tools.make_polygon(
+    [[rc - drc, rc + drc, rc + drc, rc - drc], [0, 0, 0, 0], [-drc, -drc, +drc, +drc]],
+    closed=True,
+    label="poly_enclo",
+)
+poly_coil.change_placement((BluemiraPlacement(axis=[1.0, 0.0, 0.0], angle=-90)))
+
+poly_coil.mesh_options = {"lcar": lcar_coil, "physical_group": "poly_coil"}
+coil = BluemiraFace(poly_coil)
+coil.mesh_options = {"lcar": lcar_coil, "physical_group": "coil"}
+
+# %%[markdown]
+
+# create the enclosure (rectangular cross section) and set the mesh options
+
+# %%
+poly_enclo = tools.make_polygon(
+    [[0, r_enclo, r_enclo, 0], [0, 0, 0, 0], [-r_enclo, -r_enclo, r_enclo, r_enclo]],
+    closed=True,
+    label="poly_enclo",
+)
+poly_enclo.change_placement((BluemiraPlacement(axis=[1.0, 0.0, 0.0], angle=-90)))
+
+poly_enclo.mesh_options = {"lcar": lcar_enclo, "physical_group": "poly_enclo"}
+enclosure = BluemiraFace([poly_enclo, poly_coil])
+enclosure.mesh_options = {"lcar": lcar_enclo, "physical_group": "enclo"}
+
+# %%[markdown]
+
+# create the different components
+
+# %%
+c_universe = Component(name="universe")
+c_enclo = PhysicalComponent(name="enclosure", shape=enclosure, parent=c_universe)
+c_coil = PhysicalComponent(name="coil", shape=coil, parent=c_universe)
+
+# %%[markdown]
+
+# # Mesh
+
+# Create the mesh (by default, mesh is stored in the file Mesh.msh")
+
+# %%
+
+m = meshing.Mesh()
+m(c_universe, dim=2)
+
+# %%[markdown]
+
+# Convert the mesh in xdmf for reading in fenics. Note that this requires the msh2xdmf
+# module to be available.
+
+# %%
+
+msh2xdmf.msh2xdmf("Mesh.msh", dim=2, directory=".")
+
+mesh, boundaries, subdomains, labels = msh2xdmf.import_mesh(
+    prefix="Mesh",
+    dim=2,
+    directory=".",
+    subdomains=True,
+)
+dolfin.plot(mesh)
+plt.show()
+
+# %%[markdown]
+
+# # Setup EM problem
+
+# Finally, instantiate the em solver
+
+# %%
+
+em_solver = FemMagnetostatic2d(mesh, boundaries, p=3)
+
+# %%[markdown]
+
+# Define source term (coil current distribution) for the fem problem
+
+# %%
+
+Ic = 1e6
+jc = Ic / coil.area
+markers = [labels["coil"]]
+functions = [jc]
+jtot = ScalarSubFunc(functions, markers, subdomains)
+
+# %%[markdown]
+
+# plot the source term
+# Note: depending on the geometric dimension of the coil, enclosure, and mesh
+# characteristic length, the plot could be not so "explanatory".
+
+# %%
+
+f_space = dolfin.FunctionSpace(mesh, "DG", 0)
+f = dolfin.Function(f_space)
+f.interpolate(jtot)
+dolfin.plot(f, title="Source term")
+plt.show()
+
+# %%[markdown]
+
+# solve the em problem and calculate the magnetic field B
+
+# %%
+
+em_solver.solve(jtot)
```

Comment:
The default setup is to apply Dirichlet bc = 0 to the boundary of the domain, same for the Neumann ones (not to the whole function space... DOLFIN predefines the “measures” dx, ds and dS representing integration over cells, exterior facets (that is, facets on the boundary) and interior facets, respectively).

Indeed, the default setup results in a "wrong" solution if the enclosure is too small since the "infinite" boundary conditions of the magnetic field are not respected. For this reason, in my example, the enclosure is 100 m. Of course, one can specify a smaller enclosure and appropriate boundary conditions, which is exactly the case of the fixed boundary equilibrium problem. 
Now, on the latter, we fix a constant value on the plasma separatrix, so the definition of the bc is almost simple. I have to check how the ScalarSubFunc behaves for the boundary subdomains and if the output function is well interpolated as Dirichlet boundary condition, but this is more an "exercise" for the free boundary problem (out of the scope of this PR).